source/agora/consensus/EnrollmentManager.d

```diff
@@ -711,6 +668,26 @@ public class EnrollmentManager
         return this.cycle_preimages[height];
     }
 
+    /***************************************************************************
+
+        Create signature noise for enrollment data
+
+        Parameters:
+            height = the height used for a salt value
+
+        Returns:
+            the signature noise created using the private key
+
+    ***************************************************************************/
+
+    private Pair createSignatureNoise (ulong height) nothrow
+    {
+        Pair key_pair;
+        key_pair.v = Scalar(hashMulti(this.key_pair.v, height));
```

Comment:
Whenever we derive something from something else, we usually need three component:
- The secret we derive from (here, private key);
- A constant value which state the "purpose" of the derivation;
- A non-constant value to actually make the derivation;

The first element goes without saying. Here, the third element is `height`, but it can also be a combination of other values. However we're missing the second element. Why is it important ? Because we might do the derivation in a few unrelated places, using potentially the same secret, and a data-type that might overlap. For example we could have a derivation involving the secret and a number of round. If that happens, an attacker could use the result of a derivation in the signature noise to predict the values generated for the second derivation.
Hence something like this:
```suggestion
        key_pair.v = Scalar(hashMulti(this.key_pair.v, "consensus.signature.noise", height));
```

Where we make sure to never reuse `consensus.signature.noise` would be better.